test/unittest/unittest_properties.cpp

```diff
@@ -111,6 +127,27 @@ TEST(BasicProperty, valid_p) {
     EXPECT_EQ(nntrainer::to_string(q), "this is good");
   }
 
+  { /** set -> get / to_string, dimension*/
+    DimensionOfBanana q;
+    q.set({1, 2, 3, 4});
+    EXPECT_EQ(q.get(), nntrainer::TensorDim(1, 2, 3, 4));
+    EXPECT_EQ(nntrainer::to_string(q), "1:2:3:4");
+  }
+
+  { /**< from_string -> get / to_string, dimension */
+    DimensionOfBanana q;
+    nntrainer::from_string("1:2:3:4", q);
+    EXPECT_EQ(q.get(), nntrainer::TensorDim(1, 2, 3, 4));
+    EXPECT_EQ(nntrainer::to_string(q), "1:2:3:4");
+  }
+
+  { /**< from_string -> get / to_string, dimension */
+    DimensionOfBanana q;
+    nntrainer::from_string("3:4", q);
+    EXPECT_EQ(q.get(), nntrainer::TensorDim(1, 1, 3, 4));
+    EXPECT_EQ(nntrainer::to_string(q), "1:1:3:4");
+  }
```

Comment:
can you please a couple more unittests (negative ones) - one with empty string, only ":" and more than 4 dimensions.

Reply:
i guess more than 4 dimensions is already added here - https://github.com/nnstreamer/nntrainer/pull/1161/files#diff-8bd026869e0f1961b1ba9f517c2af8ef4be981ebfd65959d1410a355d5829a9bR225

Reply:
Done.